packages/noya-state/src/reducers/pointReducer.ts

```diff
@@ -172,6 +182,168 @@ export function pointReducer(
         });
       });
     }
+    case 'setControlPointX':
+    case 'setControlPointY': {
+      if (!state.selectedControlPoint) return state;
+      const [type, amount, CanvasKit, mode] = action;
+      const controlPointType = state.selectedControlPoint.controlPointType;
```

Comment:
We should probably destructure the whole `selectedControlPoint` here, since otherwise TypeScript may think they're optional later.

```ts
const { layerId, pointIndex, controlPointType } = state.selectedControlPoint
```